src/main/java/com/mixpanel/android/mpmetrics/SystemInformation.java

```diff
@@ -122,14 +122,13 @@ public String getCurrentNetworkOperator() {
         return ret;
     }
 
-
     public Boolean isWifiConnected() {
         Boolean ret = null;
 
         if (PackageManager.PERMISSION_GRANTED == mContext.checkCallingOrSelfPermission(Manifest.permission.ACCESS_NETWORK_STATE)) {
             ConnectivityManager connManager = (ConnectivityManager) this.mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
-            NetworkInfo wifiInfo = connManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
-            ret = wifiInfo.isConnected();
+            NetworkInfo networkInfo = connManager.getActiveNetworkInfo();
```

Comment:
Why did this have to change for 23? Could a user be both on 4G and WiFi, but networkInfo.getType() returns the 4G network? Seems like explicitly requesting info for TYPE_WIFI is more correct.